sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -376,6 +376,35 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
     )
   }
 
+  test("map_concat function") {
+    val df1 = Seq(
+      (Map[Int, Int](1 -> 100, 2 -> 200), Map[Int, Int](3 -> 300, 4 -> 400)),
+      (Map[Int, Int](1 -> 100, 2 -> 200), Map[Int, Int](3 -> 300, 1 -> 400)),
+      (null, Map[Int, Int](3 -> 300, 4 -> 400))
+    ).toDF("map1", "map2")
+    checkAnswer(
+      df1.selectExpr("map_concat(map1, map2)"),
+      Seq(
+        Row(Map(1 -> 100, 2 -> 200, 3 -> 300, 4 -> 400)),
+        Row(Map(1 -> 400, 2 -> 200, 3 -> 300)),
+        Row(null)
+      )
+    )
+
+    val df2 = Seq(
+      (Map[Int, Int](1 -> 100, 2 -> 200), Map[String, Int]("3" -> 300, "4" -> 400))
+    ).toDF("map1", "map2")
+    assert(intercept[AnalysisException] {
+      df2.selectExpr("map_concat(map1, map2)").collect()
+    }.getMessage().contains("input maps of function map_concat should all be the same type"))
+    assert(intercept[AnalysisException] {
```

Comment:
can you put a blank line between tests? makes it a bit easier to see the separation.